include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -3413,9 +3415,9 @@ namespace internal
   // MatrixFree
   template <typename VectorType>
   inline void
-  check_vector_compatibility(
-    const VectorType &                            vec,
-    const internal::MatrixFreeFunctions::DoFInfo &dof_info)
+  check_vector_compatibility(const VectorType &vec,
+                             const internal::MatrixFreeFunctions::DoFInfo<
+                               typename VectorType::value_type> &dof_info)
```

Comment:
Did you run all matrix-free tests already? This might break some functionality we once had, namely reading a `double` vector with an `FEEvaluation` object of type `float`. I have not checked whether we have a test (if not, I'm to blame), but this should be easy to fix in a subsequent PR.

Reply:
Ah we do, it is (at least) `matrix_free/step-37`.

Reply:
@kronbichler Yes. : https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=2140